app/src/main/java/org/simple/clinic/bp/assignbppassport/BpPassportSheet.kt

```diff
@@ -1,13 +1,122 @@
 package org.simple.clinic.bp.assignbppassport
 
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
 import android.os.Bundle
-import org.simple.clinic.R
+import com.jakewharton.rxbinding3.view.clicks
+import io.github.inflationx.viewpump.ViewPumpContextWrapper
+import io.reactivex.Observable
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.databinding.SheetBpPassportBinding
+import org.simple.clinic.di.InjectorProviderContextWrapper
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.util.wrap
 import org.simple.clinic.widgets.BottomSheetActivity
+import javax.inject.Inject
 
-class BpPassportSheet : BottomSheetActivity() {
+class BpPassportSheet : BottomSheetActivity(), BpPassportUiActions {
+
+  companion object {
+    private const val KEY_BP_PASSPORT_NUMBER = "bpPassportNumber"
+  }
+
+  @Inject
+  lateinit var effectHandlerFactory: BpPassportEffectHandler.Factory
+
+  private lateinit var component: BpPassportSheetComponent
+
+  private lateinit var binding: SheetBpPassportBinding
+
+  private val registerNewPatientButton
+    get() = binding.registerNewPatientButton
+
+  private val addToExistingPatientButton
+    get() = binding.addToExistingPatientButton
+
+  private val events: Observable<BpPassportEvent> by unsafeLazy {
+    Observable
+        .merge(
+            registerNewPatientClicks(),
+            addToExistingPatientClicks()
+        )
+  }
+
+  private fun addToExistingPatientClicks(): Observable<BpPassportEvent> {
+    return addToExistingPatientButton
+        .clicks()
+        .map { AddToExistingPatientClicked }
+  }
+
+  private fun registerNewPatientClicks(): Observable<BpPassportEvent> {
+    return registerNewPatientButton
+        .clicks()
+        .map { RegisterNewPatientClicked(0) }
+  }
+
+  private val delegate by unsafeLazy {
+
+    MobiusDelegate.forActivity(
+        events = events,
+        defaultModel = BpPassportModel(),
+        update = BpPassportUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build(),
+    )
+  }
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    setContentView(R.layout.sheet_bp_passport)
+    binding = SheetBpPassportBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    delegate.onRestoreInstanceState(savedInstanceState)
+  }
+
+  override fun openPatientEntryScreen(bpPassportNumber: Int) {
+    val intent = Intent()
+    intent.putExtra(KEY_BP_PASSPORT_NUMBER, bpPassportNumber)
```

Comment:
Let's go with the existing search screen implementation, where we need to save the `OngoingLoginEntry` and then we can open the patient entry screen without needing to pass the BP passport identifier.